openedx/core/djangoapps/theming/tests/test_theme_style_overrides.py

```diff
@@ -35,6 +38,36 @@ def test_footer(self):
         # This string comes from header.html of test-theme
         self.assertContains(resp, "This is a footer for test-theme.")
 
+    @with_comprehensive_theme("edx.org")
+    def test_header_no_mktg_links(self):
+        """
+        Test that theme header doesn't show marketing site links for Account Settings page.
+        """
+        # Login
+        user = UserFactory.create()
+        result = self.client.login(username=user.username, password='test')
+        self.assertTrue(result)
+
+        account_settings_url = reverse('account_settings')
+        resp = self.client.get(account_settings_url)
+        self.assertEqual(resp.status_code, 200)
+        self.assertNotContains(resp, "How it Works")
+
+    @with_comprehensive_theme("edx.org")
+    def test_header_with_mktg_links(self):
+        """
+        Test that theme header show marketing site links for Profile page.
+        """
+        # Login
+        user = UserFactory.create()
+        result = self.client.login(username=user.username, password='test')
+        self.assertTrue(result)
+
+        profile_url = reverse('learner_profile', kwargs={'username': user.username})
+        resp = self.client.get(profile_url)
+        self.assertEqual(resp.status_code, 200)
+        self.assertContains(resp, "How it Works")
```

Comment:
What about the other nav elements, "Find courses" and "Schools & Partners"? This seems too brittle. Consider a helper method which performs these assertions.